_pages/guides/data-hub-framework.markdown

```diff
@@ -5,17 +5,110 @@ lead_text: ''
+
+    ```
+    mlModulePaths=src/main/ml-modules,src/main/ui-modules
+    ```
+
+5. __Only necessary for Data Hub 5.0.1__ Update the Data Hub's `build.gradle` to use a newer version of ml-gradle.  Add the `dependencies` to the `buildscript` object around line 5: 
+
+    ```JSON
+    dependencies {
+		  classpath "gradle.plugin.com.marklogic:ml-gradle:3.14.0"
+    }
+    ```
+
+6. Copy the contents of the Grove modules to the Data Hub project. Copy 
+    ```
+    grove/marklogic/ml-modules/options
+    grove/marklogic/ml-modules/root 
+    grove/marklogic/ml-modules/services
+    grove/marklogic/ml-modules/transforms
+    
+    to 
+    
+    data-hub/src/main/ui-modules
+    ```
+
+7. Copy the contents of the Grove default user profile and a dictionary document to the Data Hub project. Copy 
+    ```
+    grove/marklogic/ml-content
+    
+    to 
+    
+    data-hub/src/main
+    ```
+
+8. Edit the query options used by your Grove middle-tier's search route (by default, these query options are called `all` and are found in the `data-hub/ui-modules/options/all.xml` file) to remove the following blocks:
+
+    ```xml
+    <constraint name="eyecolor">
+      ...
+    </constraint>
+    <constraint name="docFormat">
+      ...
+    </constraint>
+    <constraint name="gender">
+      ...
+    </constraint>
+    ```
+
+9. Make any other necessary changes to the query options file. For example, you may need to remove the `<additional-query>` specifying that only docs in the `data` collection are returned.  There may be a Data Hub specific collection that is a natural fit to limit the search results for your Grove application.  
+
+10. Run `./gradlew mlLoadModules` from inside the `data-hub` directory. This command will deploy the new modules and supporting documents.  
+
+11. Delete the contents of `grove/marklogic`. 
+
+
+## Alternative Approach to Run Grove with the Data Hub
+
+In order to run a Grove UI project alongside the MarkLogic Data Hub Framework (DHF) project, we alternatively suggest creating a Grove project-specific app-server, which has its own modules database but points at the content database with the data you wish to visualize: most often the DHF project's FINAL database. The DHF project will be responsible for managing the content database (including setting up indexes - needed for facets - and security permissions), as well as the content database's related triggers and schemas databases, while the Grove project will manage the Grove project's specific app-server and modules database.
 
 Note that this implies some coordination between the DHF project and Grove. If possible, it is easiest to manage both projects within the same code repository. The Grove files should all be within a dedicated parent directory. (For example, `grove/ui`, `grove/middle-tier`, and `grove/marklogic`. Note that there is nothing magic about the `grove` directory name, so call it whatever you want.)
```

Comment:
The other approach also requires coordination. Maybe this should be a global note at the top?

Reply:
I have the same words in both areas of this document, so I think that will capture the coordination is required.

Reply:
This comment did point out DHF is still referenced.  I updated the entire document to change DHF -> Data Hub.